xmppserver/src/main/java/org/jivesoftware/openfire/http/HttpBindManager.java

```diff
@@ -756,6 +766,33 @@ protected ServletContextHandler createStaticContentHandler()
         return null;
     }
 
+    /**
+     * Adds a Jetty handler to be added to the embedded web server that is used to expose Openfire's public
+     * web-bindings (eg: BOSH / HTTP-bind and websocket).
+     *
+     * @param handler The handler (cannot be null).
+     */
+    public void addJettyHandler(final ContextHandler handler)
```

Comment:
Do you not want to use a @Nonnull check here like the removeJettyHandler(@Nonnull final ContextHandler handler) method?